tls/s2n_client_hello_retry.c

```diff
@@ -12,8 +12,32 @@
  * express or implied. See the License for the specific language governing
  * permissions and limitations under the License.
  */
+#include <stdbool.h>
 
 #include "error/s2n_errno.h"
+#include "utils/s2n_blob.h"
+#include "tls/s2n_tls.h"
+
+/* from RFC: https://tools.ietf.org/html/rfc8446#section-4.1.3*/
+/* Decimal equivalent of hex
+     CF 21 AD 74 E5 9A 61 11 BE 1D 8C 02 1E 65 B8 91
+     C2 A2 11 16 7A BB 8C 5E 07 9E 09 E2 C8 A8 33 9C
+*/
+const uint8_t hello_retry_req_random[] = {
+    207, 33, 173, 116, 229, 154, 97, 17, 190, 29, 140, 2, 30, 101, 184, 145,
+    194, 162, 17, 22, 122, 187, 140, 94, 7, 158, 9, 226, 200, 168, 51, 156
+};
```

Comment:
Here's where I meant we could just use hex.  0xCF, 0x21, 0xAD, that way it's just easier to see that the values are the same as what's in the RFC.

Reply:
To compare the two values we either have to convert the decimal to hex or reverse. We can just skip that if we have the constant's values in Decimal.
There is a unit test which do verify the pre-calculated conversion of hex to decimal.

Reply:
Why? It's just a matter of writing the number - compiler treats them the same way: https://rextester.com/MEMB66519

Reply:
ahh.. I get what do you mean. Will replace it.